include/cadmium/simulation/rt_clock/mbedclock.hpp

```diff
@@ -0,0 +1,230 @@
+/**
+ * Real-time clock based on the Driver APIs from MBED-OS version 5.
+ * Copyright (C) 2023  Ezequiel Pecker Marcosig
+ * SEDLab - University of Buenos Aires
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+#ifndef CADMIUM_SIMULATION_RT_CLOCK_MBED_HPP
+#define CADMIUM_SIMULATION_RT_CLOCK_MBED_HPP
+
+// #include <iostream>
+#include <chrono>
+#include <optional>
+#include <thread>
+#include "rt_clock.hpp"
+#include "../../exception.hpp"
+#include <mbed.h>
+
+static long MIN_TO_MICRO   = (1000*1000*60);
+static long SEC_TO_MICRO   = (1000*1000);
+static long MILI_TO_MICRO  = (1000);
+
+namespace cadmium {
+    /**
+     * Real-time clock based on the Driver APIs provided by MBED-OS version 5: https://os.mbed.com/docs/mbed-os/v5.15/apis/drivers.html.
+     * @tparam T Internal clock type. By default, it uses a double.
+     */
+    template <typename T = double>
+    class MBEDClock: RealTimeClock {
+     private:
+	// executionTimer is used in the waitUntil() to measure the elapsed time between cnsecutive calls from the Root Coordinator and the time elapsed between an event and an external interruption
+	Timer executionTimer;
+	Timeout _timeout;
+	bool expired;
+
+        // scheduler_slip = 0 if the next event (actual_delay) is in the future, AKA we are ahead of schedule
+        // scheduler_slip = <how_long_behind_schedule_we_are> if actual_delay is negative, i.e. we are behind schedule.
+        // This is then added to the next actual_delay and updated until we surpass the tolerance or recover from the slip.
+        long scheduler_slip = 0;
+
+	// double e; // elapsed time until interrupt
+
+        // 
+        /**
+         * Returns a long with the time in microseconds.
+         * @param Time value in seconds.
+         */
+        long get_time_in_micro_seconds(const double t) const {
```

Comment:
suggestion: use a notation such as `timeToMicros`, `microsToSeconds`... 
Leave the `get` and `set` words for getters and setters of the class.